### ISSUE TITLE
Auth token does not have the necessary scopes, even though I requested them. 

### ISSUE THREAD

**Issue by LBlend** [ppy/osu-web#9670]
**TLDR**: *I've managed to get a token using autherization grant but for some reason this token does not have the correct scopes, even though everything else points to it being correct.*

## Preface

I wanna start off by saying that I probably shouldn't make this issue but I see no other option of solving this for now. I'm at a point where I'm starting to believe there's something wrong on the API's end rather than me.

## The problem
I'm trying to figure out why I'm not being able to auth to `/me` in the v2 API. I think I've might have gone blind at this point for some obivious error. Can you spot it?

I start by requesting auth from a user by letting them click the following link
`https://osu.ppy.sh/oauth/authorize?client_id=2938&redirect_uri=http://localhost:6969/callback&state=INSERTSTATEHERE&response_type=code&scope=identify`

...which is successful, and I get the long code back
 
I then request for a token using the code (github thinks it's misformatted due to the `:` in my redirect uri but that shouldn't be an issue in practice)
```json
{'client_id': 2938, 'client_secret': 'insert_epic_secret_here', 'grant_type': 'authorization_code', 'redirect_uri': 'http://localhost:6969/callback', 'scope': 'identify', 'code': 'insert_long_code_here'}
```

...which is also successful. I get a token back.

But then when I try to request some data from the `/me` endpoint my request fails
```
> GET /api/v2/users/me/osu HTTP/2
> Host: osu.ppy.sh
> content-type: application/json
> authorization: Bearer INSERTLONGTOKENHERE
```
```json
{
    "error": "A required scope is missing."
}
```

there seems to be a scope issue with my token. I've tried payloading the scope on to the last request (even though I'm not supposed to do that according to the docs). 

Looking at my authenticated clients though, it should have the correct scopes
![image](https://user-images.githubusercontent.com/24893890/209246383-bb0fc8e7-d1b3-4edb-b71e-8dee56b2f6af.png)

Do you have any idea what might've caused this problem?

**Comment by tybug** [comment]
you're requesting against the [`/api/v2/users/{id}`](https://osu.ppy.sh/docs/index.html#get-user) endpoint, which requires the `public` scope. You're looking for the [`/api/v2/me`](https://osu.ppy.sh/docs/index.html#get-own-data) endpoint.

**Comment by LBlend** [comment]
Turns out I really am blind! How embarassing!

Thanks for the help! 😅
